cmake/patch_repo.cmake

```diff
@@ -0,0 +1,16 @@
+
+# Function to generate a PATCH_COMMAND, calling the
+# patch_repo.py script using a target set of patches.
+
+function(get_patch_command patch_dir patch_command_out)
+    set(patch_script ${CMAKE_CURRENT_LIST_DIR}/patch_repo.py)
```

Comment:
I think that using `${CMAKE_CURRENT_LIST_DIR}` in this context means it won't be evaluated until the function is called, and then it'll use its caller's value of that variable. So if a function _not_ in the `cmake` subdir were to call `get_patch_command`, it would get the wrong directory here.

At the moment this is fine, because all the callers live in that subdir too, but I could imagine it maybe not staying that way. So for one extra safety precaution, how about copying `${CMAKE_CURRENT_LIST_DIR}` into a new variable in this include file:
```cmake
set(patch_list_dir ${CMAKE_CURRENT_LIST_DIR})
function(get_patch_command patch_dir patch_command_out)
    set(patch_script ${patch_list_dir}/patch_repo.py)
```
and similarly refer to `${patch_list_dir}/../patches`?

Reply:
True - although setting `patch_list_dir` inside `patch_repo.cmake` does not guarantee that the value will still be correct by the time the function is called, since it can be freely modified within the scope. The safest option I can see is pass the toolchain root to the function as an argument, which can then be used within to derive the patch/script locations.

Reply:
I'm not sure what you mean by that. Yes, something else _could_ modify `patch_list_dir`, but it would have to do it on purpose – I can't see any plausible way it would happen by accident, in the way that "oops I called `get_patch_command` from a script somewhere other than the `cmake` directory" seemed to me like an easy mistake.

But this version works too!